### PR TITLE
Implemented system to train on a subset of the dataset, favoring higher rated images

### PR DESCRIPTION
**Pull request by JanGerritsen**
First: I agree to the Contributor License Agreement

This is the second part of the dynamization of training data through a training session.
This PR focuses on skewing the selection of images to train to higher rated images.

This is done by leaving out less favorabel images in later epochs, at an increasing rate.
The result is that the model has seen more favorabel images more often and more recently at the end of the training session.

To accomplish this the user can define a rating per image in the YAML file.
The rating of one image is compared to the ratings of the other images. A rating twice as high then the rating of another image would result in twice as high a chance to be picked in the later stages of the training.

The second parameter is the target dropout rate in percent at the last epoch of the training. 
The default is 50%, meaning that 50 percent of the training images will be skipped at the last epoch.

The first epoch will always be trained on all images. The epochs between will drop out more and more images in a linear regression.

Each image has a chance to be picked for an epoch, but the chance of being picked relates to its rating.

### CODE REVIEW

**Comment by SunShopsAdam** [comment]
Could you please be specific about how the ratings are read from the yaml file? It doesn't make a lot of sense to me to have the yaml contain ratings per image, when the rating could be read from the captions.

**Comment by SunShopsAdam** [comment]
Oh I see. It's part of the YAML tagging per image and not the training session YAML file. So the yaml just includes 
main_prompt: A portrait of Cloud Strife
tags:
  - tag: low angle shot
  - tag: looking to the side
  - tag: holding buster sword
    weight: 1.5
  - tag: clouds in background
    weight: 0.5
  - tag: smiling
    weight: 0.8
max_caption_length: 1024
rating: 1.0

Am I understanding that correctly?
